File: DSA/patterns/frequency_counter/FindingDuplicates.js
// FindingDuplicates.js

function hasDuplicates(arr) {
  /**
   * Check if an array contains duplicates using the Frequency Counter pattern.
   */
  let freqCount = {};

  for (let num of arr) {
    if (freqCount[num]) {
      return true;
    }
    freqCount[num] = 1;
  }

  return false;
}

File: DSA/patterns/frequency_counter/finding_duplicates.py
"""finding_duplicates.py"""

def has_duplicates(arr):
    """
    Check if an array contains duplicates using the Frequency Counter pattern.
    """
    freq_count = {}

    for num in arr:
        if num in freq_count:
            return True
        freq_count[num] = 1

    return False
